packages/colorpickers/src/elements/ColorPicker/reducer.ts

```diff
@@ -133,73 +172,136 @@ export const reducer: ReducerType = (state, action) => {
         };
       }
 
-      return state;
+      return {
+        ...state,
+        hexInput: action.payload,
+        color
+      };
     }
     case 'RED_CHANGE': {
-      const red = Number(action.payload);
+      let red = parseInt(action.payload, 10);
```

Comment:
`parseInt` is a more specific type of number parsing that will help remove invalid notation.